mongosql/src/air/desugarer/remove_id.rs

```diff
@@ -28,6 +28,6 @@ impl Visitor for RemoveIdDesugarerVisitor {
                 .insert(id, ProjectItem::Exclusion)
                 .unwrap();
         }
-        node
+        node.walk(self)
```

Comment:
The comment at the top of this file says "This removes _id using the last Project stage in the pipeline". Can we update that to accurately reflect what we are doing now?

For Nathan: it seems at some point we changed our translations to go from excluding `_id` in every `$project` stage to only doing it in the _last_ `$project` stage. Excluding in only the last `$project` breaks some of the critical assumptions we relied on for our `DISTINCT` implementations. Chiefly, that only the specifically selected fields would be projected within the pipeline. This change is necessary to restore that invariant. The good news is that an exclusion is very cheap (and the "unnecessary" further ones down the pipeline should even be free).